tests/test_fastzipfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+import shutil
+import subprocess
+import sys
+import time
+import tempfile
+import zipfile
+
+import fastzipfile
+
+try:
+    import pyminizip
+except ImportError:
+    pyminizip = None  # type: ignore
+
+password = "foo"
+
+
+def create_encrypted_test_file(path):
+    old_working_directory = os.getcwd()
+
+    files = {
+        "random-8MiB": os.urandom(8 * 1024 * 1024),
+        "digits-8MiB": b'01234567' * (1024 * 1024),
+    }
+
+    try:
+        folder = tempfile.TemporaryDirectory()
+        os.chdir(folder.name)
+        for name, contents in files.items():
+            with open(name, "wb") as file:
+                file.write(contents)
+
+        if os.path.exists(path):
+            os.remove(path)
+
+        command = []
+        if shutil.which("zip"):
```

Comment:
Why not use `zipfile` module to create password protected zip for testing instead of relying on external executable?

Reply:
Becuase `zipfile` can [only decompress](https://docs.python.org/3/library/zipfile.html#zipinfo-objects), not compress, at least that's what the manual stats. I did not try it:

>  It supports decryption of encrypted files in ZIP archives, but it currently cannot create an encrypted file. Decryption is extremely slow as it is implemented in native Python rather than C.